id3v2frames.go

```diff
@@ -67,6 +71,26 @@ func encodingDelim(enc byte) ([]byte, error) {
 	}
 }
 
+func dataSplit(b []byte, enc byte) ([][]byte, error) {
+	delim, err := encodingDelim(enc)
+	if err != nil {
+		return nil, err
+	}
+	result := bytes.SplitN(b, delim, 2)
+
+	if len(result) <= 1 {
+		return result, nil
+	}
+
+	if result[1][0] == 0 {
+		// there was a double (or triple) 0 and we cut too early
+		result[0] = append(result[0], make([]byte, 0)...)
```

Comment:
yes, the idea was to add result[0] = append(result[0], byte(0)
We have a case when the 0 is at result[1][0] instead being at the end of result[0]